### DISCUSSION TITLE
Single key application, key got banned

### DISCUSSION THREAD

**Discussion by Kuba314** [HypixelDev/PublicAPI#298] [Q&A]
I have an application hosted on Amazon AWS and I used more keys for my application and the hypixel abuse system detected it and probably put the IP on a blacklist or something. But when I ran the application again, this time with only a single key, it still got banned.

The application is not breaking any rules right now. What do I do to be able to send requests without getting banned again?

**Comment by GoogleSites** [comment]
Do you use the `X-Ratelimit-Reset` and `X-Ratelimit-Remaining` headers to avoid going over the ratelimit? If not, the IP (or block of IPs) where your machine is hosted may have been flagged/banned by Hypixel or something that Hypixel relies on (Cloudflare?).

**Comment by Kuba314** [comment]
Didn't know those header existed, but I limit my requests and none of them went over the limit. The IP definitely got blacklisted, because all of the keys that were used got invalidated.

So... 1 key, under 120 requests/min. Still got invalidated. Why?

Edit: now I'm running on 3 requests/min to not trigger the system, because I got tired of resetting my key.
Edit: how did you find out that those headers exist? didn't find them in the docs earlier. Is it something you found out just by looking at the headers?

**Comment by GoogleSites** [comment]
It seems weird that your key would get invalidated so consistently. I used 3 keys (maximum for an application without it being abusive) at 360 req/min for ~a week straight and nothing got flagged on my end. If your 3 req/min doesn't flag anything, then maybe you were going over the limit? The request reset isn't at x:00 of every minute

They don't mention them anywhere but they're fairly common headers in APIs so I just checked if they were there.

**Comment by Kuba314** [comment]
I used 7 keys.
> 3 keys (maximum for an application without it being abusive)

Where did you get this info from?


> for ~a week straight and nothing got flagged on my end

I used 6 keys for more than 2 weeks and nothing happened. Even with occasional throttles caused by requests getting delayed by more than 5 seconds and screwing with my rate limitter.


> then maybe you were going over the limit?

I used a rate limitter that executed 113 requests every 65 seconds with a time difference of 0.15 seconds. There is no reason for the requests to get throttled. They'd have to get delayed by 5+ seconds to throttle.

**Comment by robere2** [comment]
Just to confirm, you have tried generating new keys, correct? In the past Hypixel has deleted keys, but I'm not aware of IPs being blacklisted outside of sending too many requests with invalid API keys. If your program was running for a considerable amount of time with invalid API keys, this could be why your IP was blocked, in which case you'd need to contact support.

**Comment by Kuba314** [comment]
Yes, I tried regenerating new keys. The new keys work just fine. Well... they do, but only for like half an hour.

Some of the keys I used like 3 months ago got invalidated because of an error in my code causing throttles. This has been fixed for a long time. The new ones definitely weren't throttling.
